http-common/src/connector.rs

```diff
@@ -768,8 +754,56 @@ fn get_num_systemd_sockets() -> Result<Option<std::os::unix::io::RawFd>, String>
         }
     }
 
-    // Return the number of systemd sockets.
-    Ok(Some(listen_fds))
+    //If there is only one socket, we know this is the identity service which uses only one socket, so we have a match:
+    if listen_fds == 1 {
+        return Ok(Some(SD_LISTEN_FDS_START as i32));
+    }
+
+    //if there is more than 1 and we don't have a socket name to match, this is edged telling us that there is no systemd socket we can match.
+    let socket_name = match socket_name {
+        Some(socket_name) => socket_name,
+        None => return Ok(None),
+    };
+
+    // If there is more than one socket, this is edged. We can attempt to match the socket name to systemd.
+    // This happens when a unix Uri is provided in the config.toml. Systemd sockets get created noneless so we still prefer to use them.
+    // If no socket name is provided, this mean edged is telling us there is not matching systemd socket, we return early. See above "((listen_fds > 1)&&(socket_name == None))"
+    // If a socket name is provided but we don't see the env variable LISTEN_FDNAMES, it means we are probably on an older OS, we can't match either.
+    let listen_fdnames = match get_env("LISTEN_FDNAMES")? {
+        Some(listen_fdnames) => listen_fdnames,
+        None => return Ok(None),
+    };
+    let listen_fdnames: Vec<&str> = listen_fdnames.split(':').collect();
+
+    #[allow(
+        clippy::cast_possible_truncation,
+        clippy::cast_sign_loss,
+        clippy::cast_possible_wrap
+    )]
+    if listen_fds != listen_fdnames.len() as i32 {
```

Comment:
Would be better to use `try_into()` instead of a cast and remove the clippy allows.

Reply:
hum, it is not happy. But I will use "std::os::unix::io::RawFd" similar to above.

Reply:
found a way